From 78a6cb5304be1ff5c6e4b274ea9711745a28ec24 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Benno=20F=C3=BCnfst=C3=BCck?= <benno.fuenfstueck@gmail.com>
Date: Wed, 1 Nov 2017 06:54:40 +0100
Subject: [PATCH] fix live updates

---
 main.coffee | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/main.coffee b/main.coffee
index 57f3b09..f61e6dd 100644
--- a/main.coffee
+++ b/main.coffee
@@ -255,8 +255,8 @@ etherpad.stderr.on 'data', (line) ->
 
 wss = new WebSocketServer {server:server}
 wss.broadcast = (msg, exclude, scope=null) ->
-  for c in this.clients
-    unless c.authenticated then continue
+  this.clients.forEach (c) ->
+    unless c.authenticated then return
     if c isnt exclude and (scope is null or scope is c.authenticated.scope)
       try
         c.send msg
@@ -282,7 +282,7 @@ wss.on 'connection', (sock) ->
                 sock.send JSON.stringify {type: 'assign', subject: i.challenge, data: [{name: i.user}, true]}
             # notify all users about new authentication and notify new socket about other users
             wss.broadcast JSON.stringify {type: 'login', data: ans.name}
-            for s in wss.getClients()
+            wss.getClients().forEach (s) ->
               if s.authenticated and s.authenticated.name isnt ans.name
                 sock.send JSON.stringify {type: 'login', data: s.authenticated.name}
     else
@@ -304,7 +304,7 @@ wss.on 'connection', (sock) ->
             db.modifyChallenge c.id, c.title, c.category, c.points
           else
             db.addChallenge msg.data.ctf, c.title, c.category, c.points
-        for s in wss.clients
+        wss.clients.forEach (s) ->
           if s.authenticated and s.authenticated.scope is msg.data.ctf
             s.send JSON.stringify {type: 'ctfmodification'}
       else console.log msg
-- 
2.14.3

